core/src/main/java/com/novoda/noplayer/internal/exoplayer/ExoPlayerFacade.java

```diff
@@ -66,14 +70,67 @@ boolean isPlaying() {
         return exoPlayer != null && exoPlayer.getPlayWhenReady();
     }
 
+    boolean isPlayingAdvert() {
+        return isPlaying() && exoPlayer.isPlayingAd();
+    }
+
+    boolean isPlayingContent() {
+        return isPlaying() && !exoPlayer.isPlayingAd();
+    }
+
+    long mediaDurationInMillis() throws IllegalStateException {
+        assertVideoLoaded();
+        return exoPlayer.getDuration();
+    }
+
     long playheadPositionInMillis() throws IllegalStateException {
         assertVideoLoaded();
         return exoPlayer.getCurrentPosition();
     }
 
-    long mediaDurationInMillis() throws IllegalStateException {
+    long advertBreakDurationInMillis() {
         assertVideoLoaded();
-        return exoPlayer.getDuration();
+        Timeline currentTimeline = exoPlayer.getCurrentTimeline();
+        if (!isPlayingAdvert() || adsLoader.isAbsent() || !(currentTimeline instanceof SinglePeriodAdTimeline)) {
+            return 0;
+        }
+        SinglePeriodAdTimeline adTimeline = (SinglePeriodAdTimeline) currentTimeline;
+        Timeline.Period period = adTimeline.getPeriod(0, new Timeline.Period());
+
+        int currentAdGroupIndex = exoPlayer.getCurrentAdGroupIndex();
+        int advertCount = period.getAdCountInAdGroup(currentAdGroupIndex);
+
+        long advertBreakDurationInMicros = combinedAdvertDurationInGroup(period, advertCount);
+        return C.usToMs(advertBreakDurationInMicros);
+    }
+
+    long positionInAdvertBreakInMillis() {
+        assertVideoLoaded();
+        Timeline currentTimeline = exoPlayer.getCurrentTimeline();
+        if (!isPlayingAdvert() || adsLoader.isAbsent() || !(currentTimeline instanceof SinglePeriodAdTimeline)) {
+            return 0;
+        }
+        SinglePeriodAdTimeline adTimeline = (SinglePeriodAdTimeline) currentTimeline;
+        Timeline.Period period = adTimeline.getPeriod(0, new Timeline.Period());
+
+        int advertCount = exoPlayer.getCurrentAdIndexInAdGroup();
+
+        long playedAdvertBreakDurationInMicros = combinedAdvertDurationInGroup(period, advertCount);
+        return C.usToMs(playedAdvertBreakDurationInMicros) + playheadPositionInMillis();
+    }
+
+    private long combinedAdvertDurationInGroup(Timeline.Period period,
```

Comment:
fits in one line

Reply:
https://github.com/novoda/no-player/pull/208/commits/bc9549fec4b44fc13816f97e8cad00ca8a00b009